projects/hathor-wallet-headless/hsm-c-bridge.md

```diff
@@ -0,0 +1,345 @@
+
+### Code compilation
+After that, for UNIX-like systems, compile the generated file using a command similar to this sample below ( generating a `mainExc` executable from a `main.c` file and linking with the Dinamo shared libraries:
+> `gcc -o mainExc main.c -ltacndlib`
+
+Another more robust approach is to use [CMake](https://cmake.org/), generating a makefile based on the local operating system configurations.
+
+This compilation would be necessary for each kind of environment where the Headless Wallet will be run. For this first version, our focus will be compiling it to [run on Alpine Linux 3.14](https://github.com/HathorNetwork/hathor-wallet-headless/blob/master/Dockerfile), the version used by [our public dockerized application](https://hub.docker.com/r/hathornetwork/hathor-wallet-headless).
+
+### Code Delivery
+On environments where this HSM communication will happen, the executables should be moved to the main Headless Wallet application folder, to ensure they will be accessible by the running Node.js process within its docker environment.
+
+Since the usage of this kind of device is a niche of large corporate partners, this should be implemented initially in a dedicated `Dockerfile` for HSM-enabled clients, ensuring a reproducible environment for both tests and production code.
+
+### Sample code
+Sample code for signing a transaction:
+```c
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "dinamo.h" /* Dinamo headers */
+#include "htrUtils.h" /* Hathor headers */
+
+int main(int argc, char **argv)
+{
+  // Return value of the latest operation
+  int nRet = 0;
+  struct AUTH_PWD authPwd;
+  HSESSIONCTX hSession = NULL;
+
+  // XPriv Key name
+  char szId[] = HTR_TEST_XPRIV_KEYNAME;
+
+  BYTE ver = DN_BCHAIN_VER_WIF_TEST_NET;
+
+  // Address to derive child priv key from
+  int nChild = 0;
+
+  // Child Priv Key Name
+  char szDest[MAX_OBJ_ID_FQN_LEN] = "HTR_CHILD_KEY";
+  DN_BCHAIN_KEY_INFO pbKeyInfo;
+
+  // Message buffer
+  BYTE pbMsg[1024] = {0};
+
+  // Output encoding buffer
+  char pbOutputEncode[1024 * 2 + 1];
+  DWORD dwOutputLength = sizeof(pbOutputEncode);
+
+  // Hash buffer
+  BYTE pbHash[DN_BCHAIN_HASH_KECCAK256_LEN] = {0};
+  DWORD dwHashLen = sizeof(pbHash);
+
+   // Signature buffer
+  BYTE pbSig[DN_BCHAIN_MAX_SIG_LEN] = {0};
+  DWORD dwSigLen = sizeof(pbSig);
+
+  // Public Key retrieval
+  DN_BCHAIN_PBK pPbk = {0};
+
+  //=======================================================================
+  // Command Line Input Validation example, for alternative implementations
+  //=======================================================================
+  if (argc < 3) {
+    printf("Missing input parameters\n");
+    printf("Usage: signOnPath path {number}\n");
+    return 1;
+  }
+  nChild = atoi(argv[2]);
```

Comment:
I know this is not the actual code that will be used, but we should somehow verify user input.